Code/Framework/AzFramework/Platform/Linux/AzFramework/Process/ProcessCommunicator_Linux.cpp

```diff
@@ -178,61 +238,104 @@ namespace AzFramework
+        CommunicatorDataImpl* platformImpl = static_cast<CommunicatorDataImpl*>(m_communicatorData.get());
+        platformImpl->m_stdInAndOutPollHandle = epoll_create1(0);
+
+        AZ_Assert(platformImpl->m_stdInAndOutPollHandle != -1, "Failed to create epoll handle to monitor output process");
+        if (platformImpl->m_stdInAndOutPollHandle == -1) 
+        {
+            processData->m_startupInfo.CloseAllHandles();
+            CloseAllHandles();
+            return false;
+        }
+        
+        struct epoll_event event;
+        event.events = EPOLLIN;
+        event.data.fd = m_stdOutRead->GetHandle();
+        if(epoll_ctl(platformImpl->m_stdInAndOutPollHandle, EPOLL_CTL_ADD, event.data.fd, &event))
+        {
+            AZ_Assert(false, "Unable to add a file descriptor for stdOut to epoll");
+            processData->m_startupInfo.CloseAllHandles();
+            CloseAllHandles();
+            return false;
+        }
+
+        event.events = EPOLLIN;
+        event.data.fd = m_stdErrRead->GetHandle();
+        if(epoll_ctl(platformImpl->m_stdInAndOutPollHandle, EPOLL_CTL_ADD, event.data.fd, &event))
+        {
+            AZ_Assert(false, "Unable to add a file descriptor for stdErr to epoll");
+            processData->m_startupInfo.CloseAllHandles();
+            CloseAllHandles();
+            return false;
+        }
 
         m_initialized = true;
         return true;
     }
 
-    void StdInOutProcessCommunicator::WaitForReadyOutputs(OutputStatus& status) const
+    void StdInOutProcessCommunicator::WaitForReadyOutputs(OutputStatus& status)
     {
+        using namespace ProcessCommunicatorLinuxInternal;
+
         status.outputDeviceReady = m_stdOutRead->IsValid() && !m_stdOutRead->IsBroken();
         status.errorsDeviceReady = m_stdErrRead->IsValid() && !m_stdErrRead->IsBroken();
         status.shouldReadOutput = status.shouldReadErrors = false;
 
+        CommunicatorDataImpl* platformImpl = static_cast<CommunicatorDataImpl*>(m_communicatorData.get());
         if (status.outputDeviceReady || status.errorsDeviceReady)
         {
-            fd_set readSet;
-            int maxHandle = 0;
-
-            FD_ZERO(&readSet);
-
-            if (status.outputDeviceReady)
-            {
-                int currentHandle = m_stdOutRead->GetHandle();
-                FD_SET(currentHandle, &readSet);
-                maxHandle = currentHandle > maxHandle ? currentHandle : maxHandle;
-            }
-
-            if (status.errorsDeviceReady)
-            {
-                int currentHandle = m_stdErrRead->GetHandle();
-                FD_SET(currentHandle, &readSet);
-                maxHandle = currentHandle > maxHandle ? currentHandle : maxHandle;
-            }
-
-            if (select(maxHandle + 1, &readSet, nullptr, nullptr, nullptr) != -1)
+            struct epoll_event events[2];
+            if (int numberSignalled = epoll_wait(platformImpl->m_stdInAndOutPollHandle, events, 1, -1) > -1)
```

Comment:
I don't think this function actually needed to be changed from `const`

A `const CommunicatorDataImpl*` could still access the `m_stdInAndOutPollHandle` int and work just fine.

Reply:
it was changed due to windows, actually.  Windows uses Poll(Handle) which is non const cuz it could do handle->Break().

Reply:
Ok, thanks for the clarification.